lunatrace/bsl/ingest-worker/pkg/vulnerability/advisory/osv.go

```diff
@@ -334,13 +350,64 @@ func mapAffectedElement(element schema.OsvSchemaAffectedElem) *gql.Vulnerability
 		On_conflict: VulnerabilityPackageOnConflict,
 	}
 
-	return &gql.Vulnerability_affected_insert_input{
+	queryBody := &gql.Vulnerability_affected_insert_input{
 		Affected_range_events: rangeEvents,
 		Affected_versions:     mapAffectedVersion(element.Versions),
 		Database_specific:     util.Ptr(databaseSpecific),
 		Ecosystem_specific:    util.Ptr(ecosystemSpecific),
 		Package:               affectedPackage,
 	}
+
+	if len(element.Ranges) != 0 {
+		queryBody.Ranges = buildRangeInsert(element.Ranges)
```

Comment:
This is the part im least sure about.  It was throwing errors on insert if this key was in the struct so I decided to just not add it unless there are any ranges to work with. Im sure theres a better way, was just stumbling through